orchestrator/careplanservice/handle_createtask.go

```diff
@@ -227,15 +234,19 @@ func (s *Service) handleCreateTask(ctx context.Context, request FHIRHandlerReque
 					Type:      to.Ptr("Task"),
 				},
 			})
-			tx.Update(carePlan, "CarePlan/"+*carePlan.Id)
+
+			updated, err := careteamservice.Update(s.fhirClient, *carePlan.Id, task, tx)
+			if err != nil {
+				return nil, fmt.Errorf("failed to update CarePlan: %w", err)
+			}
+
+			if !updated {
```

Comment:
this looks like the the wrong negation? No tests failing on this?

Reply:
This looks weird but is actually correct. The `careteamservice` already adds the CarePlan to the transaction but only if it was updated. If it's not updated we still want to update the careplan (since the activity changed)